CMakeLists.txt

```diff
@@ -292,22 +338,113 @@ else()
+    PUBLIC cxx_std_11
+)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    target_compile_options(openPMD.ADIOS1.Serial PUBLIC "/bigobj")
+    target_compile_options(openPMD.ADIOS1.Parallel PUBLIC "/bigobj")
+endif()
+if(openPMD_USE_INTERNAL_VARIANT)
+    target_include_directories(openPMD.ADIOS1.Serial SYSTEM PUBLIC
+        $<BUILD_INTERFACE:${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/variant/include>)
+    target_include_directories(openPMD.ADIOS1.Parallel SYSTEM PUBLIC
+        $<BUILD_INTERFACE:${openPMD_SOURCE_DIR}/share/openPMD/thirdParty/variant/include>)
+else()
+    target_link_libraries(openPMD.ADIOS1.Serial PUBLIC mpark_variant)
+    target_link_libraries(openPMD.ADIOS1.Parallel PUBLIC mpark_variant)
+endif()
+
+target_include_directories(openPMD.ADIOS1.Serial SYSTEM PRIVATE
+    ${openPMD_SOURCE_DIR}/include)
+target_include_directories(openPMD.ADIOS1.Parallel SYSTEM PRIVATE
+    ${openPMD_SOURCE_DIR}/include)
+
+if(openPMD_HAVE_MPI)
+    target_link_libraries(openPMD.ADIOS1.Parallel PUBLIC MPI::MPI_C MPI::MPI_CXX)
+    target_compile_definitions(openPMD.ADIOS1.Parallel PRIVATE "-DopenPMD_HAVE_MPI=1")
+else()
+    target_compile_definitions(openPMD.ADIOS1.Parallel PRIVATE "-DopenPMD_HAVE_MPI=0")
+endif()
 if(openPMD_HAVE_ADIOS1)
-    target_link_libraries(openPMD PUBLIC ${ADIOS_LIBRARIES})
-    target_include_directories(openPMD SYSTEM PUBLIC ${ADIOS_INCLUDE_DIRS})
+    set_target_properties(openPMD.ADIOS1.Serial PROPERTIES
+        CXX_EXTENSIONS OFF
+        CXX_STANDARD_REQUIRED ON
+        POSITION_INDEPENDENT_CODE ON
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN ON
+    )
+    if("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
+        set_target_properties(openPMD.ADIOS1.Serial PROPERTIES
+            LINK_FLAGS "-Wl,--exclude-libs,ALL")
+    elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Darwin")
+        set_target_properties(openPMD.ADIOS1.Serial PROPERTIES
+            XCODE_ATTRIBUTE_STRIP_STYLE "non-global"
+            XCODE_ATTRIBUTE_DEPLOYMENT_POSTPROCESSING "YES"
+            XCODE_ATTRIBUTE_SEPARATE_STRIP "YES"
+        )
+    endif()
+    foreach(adlib ${ADIOS_LIBRARIES_SEQUENTIAL})
+        target_link_libraries(openPMD.ADIOS1.Serial PRIVATE ${adlib})
+    endforeach()
+    target_include_directories(openPMD.ADIOS1.Serial SYSTEM PRIVATE ${ADIOS_INCLUDE_DIRS_SEQUENTIAL})
+    target_compile_definitions(openPMD.ADIOS1.Serial PRIVATE "${ADIOS_DEFINITIONS_SEQUENTIAL}")
+    target_compile_definitions(openPMD.ADIOS1.Serial PRIVATE "-DopenPMD_HAVE_ADIOS1=1")
+    target_compile_definitions(openPMD.ADIOS1.Serial PRIVATE "-DopenPMD_HAVE_MPI=0")
+    target_compile_definitions(openPMD.ADIOS1.Serial PRIVATE "-D_NOMPI=1")
+
+    if(openPMD_HAVE_MPI)
+        set_target_properties(openPMD.ADIOS1.Parallel PROPERTIES
+            CXX_EXTENSIONS OFF
+            CXX_STANDARD_REQUIRED ON
+            POSITION_INDEPENDENT_CODE ON
+            CXX_VISIBILITY_PRESET "hidden"
```

Comment:
Why a string here and a raw literal for ```openPMD.ADIOS1.Serial```?

Reply:
That's unfortunately syntax in CMake to set `-fvisibility-inlines-hidden` (or equivalent)

Reply:
Ah, now I see. yes, we can skip the quotes. Updated.